### PR TITLE
Retry on HTTP client errors

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Retry the request when the HTTP client returns an error.
Fixed block blob test.
Remove container prefix to allow more unique names.
Fixed tests that stopped recording prematurely.
Update recordings.

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.
 
[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/master/CHANGELOG.md

### CODE REVIEW

**Comment by jhendrixMSFT** [comment]
Fixes https://github.com/Azure/azure-sdk-for-go/issues/14488

File: storage/client.go [hunk 1]
```diff
@@ -107,7 +107,7 @@ func (ds *DefaultSender) Send(c *Client, req *http.Request) (resp *http.Response
 			return resp, err
 		}
 		resp, err = c.HTTPClient.Do(rr.Request())
-		if err != nil || !autorest.ResponseHasStatusCode(resp, ds.ValidStatusCodes...) {
+		if err == nil && !autorest.ResponseHasStatusCode(resp, ds.ValidStatusCodes...) {
```
Comment (catalinaperalta) [inline storage/client.go:110]: This is what will allow retrying on errors?

File: storage/container_test.go [hunk 2]
```diff
@@ -181,12 +181,12 @@ func (s *ContainerSuite) TestCreateContainerIfNotExists(c *chk.C) {
...
 func (s *ContainerSuite) TestCreateContainerIfExists(c *chk.C) {
 	cli := getBlobClient(c)
+	rec := cli.client.appendRecorder(c)
+	defer rec.Stop()
+
 	cnt := cli.GetContainerReference(containerName(c))
 	cnt.Create(nil)
 	defer cnt.Delete(nil)
-	rec := cli.client.appendRecorder(c)
-	cnt.bsc = &cli
```
Comment (catalinaperalta) [inline storage/container_test.go:188]: You're not assigning this line anymore....not sure if you need it?

File: storage/directory_test.go [hunk 3]
```diff
@@ -125,10 +128,6 @@ func (s *StorageDirSuite) TestCreateDirectoryIfExists(c *chk.C) {
 	dir := root.GetDirectoryReference("dir")
 	dir.Create(nil)
 
-	rec := cli.client.appendRecorder(c)
-	dir.fsc = &cli
```
Comment (catalinaperalta) [inline storage/directory_test.go:129]: Same question as above

File: storage/client.go (hunk 1 above)
Comment (jhendrixMSFT) [reply storage/client.go:110]: Yep!  It's pretty much the same thing we do in our retry policy.

File: storage/container_test.go (hunk 2 above)
Comment (jhendrixMSFT) [reply storage/container_test.go:188]: Correct, it's not used so I removed it.

**Review by catalinaperalta — APPROVED** [review]

**Pull request merged**